app-frontend/src/app/core/services/auth.service.js

```diff
@@ -28,7 +32,20 @@ export default (app) => {
                 if (error) {
                     return;
                 }
+
                 this.store.set('profile', profile);
+                this.featureFlagOverrides.setUser(profile.user_id);
```

Comment:
The system of overrides and which ones persist across reloads is a bit complicated so it might be good to have some documentation of it, either as a docstring here or maybe on the `featureFlagOverrides` service, that included:
- Places where feature flags (and overrides) can be set
- What the precedence is for each type of override
- Which ones are expected to persist across reloads.

Most of that info is in the PR text already so a lot of it could be copy-pasted from there.